soccer/planning/InterpolatedPath.cpp

```diff
@@ -366,11 +366,91 @@ bool Planning::InterpolatedPath::evaluate(float t, Geometry2d::Point &targetPosO
 	return true;
 }
 
+size_t Planning::InterpolatedPath::size() const
+{
+	 return points.size();
+}
+
+bool Planning::InterpolatedPath::valid() const 
+{ 
+	return !points.empty(); 
+}
+
 float Planning::InterpolatedPath::getTime(int index) const
 {
-	if (maxSpeed == -1 || maxAcceleration == -1) {
-        throw std::runtime_error("You must set maxSpeed and maxAcceleration before calling Path.evaluate()");
-    }
+	return times[index];
+}
 
-	return Trapezoidal::getTime(length(0,index), length(), maxSpeed, maxAcceleration, startSpeed, endSpeed);
+
+float Planning::InterpolatedPath::getDuration() const
+{
+	if (times.size()>0) {
+		return times.back();
+	} else {
+		return 0;
+	}
+}
+
+unique_ptr<Path> Planning::InterpolatedPath::subPath(float startTime, float endTime) const
+{
+	if (endTime > 0 && startTime > endTime) {
+		throw invalid_argument("startTime can't be after endTime");
+	}
+
+	if (startTime<0) {
+		throw invalid_argument("startTime can't be less than zero");
+	}
+
+	int start = 0;
+	while(times[start]<startTime) {
```

Comment:
Won't this get an index out of bounds in the case that startTime is greater than the duration of the path?